pkg/controller/node/node_controller.go

```diff
@@ -363,6 +364,12 @@ func (ctrl *Controller) makeMasterNodeSchedulable(node *corev1.Node) error {
 		if _, hasWorkerLabel := newLabels[WorkerLabel]; !hasWorkerLabel {
 			newLabels[WorkerLabel] = ""
 		}
+		// Legacy clusters do not have the node-role.kubernetes.io/control-plane
```

Comment:
the `makeMasterNodeSchedulable` seems to be called for clusters where the control plane nodes acts as workers. So for most of the clusters, I don't think it would even hit this logic?

Maybe we can lift this up to `reconcileMaster()` instead and have it check every time it re-checks the control plane nodes?

Reply:
I have an idea on a patch. I'll update this PR.

Reply:
I updated the PR to add the logic in reconcileMaster() and perform the update.